gateway-provider-security-jwt/src/test/java/org/apache/knox/gateway/provider/federation/TokenIDAsHTTPBasicCredsFederationFilterTest.java

```diff
@@ -372,6 +372,11 @@ public void start() throws ServiceLifecycleException {
         public void stop() throws ServiceLifecycleException {
         }
 
+        @Override
+        public boolean storeTokensInMemoryOnly() {
```

Comment:
Rather than adding this method to all the implementations, I tend to think of these things in terms of  interfaces (e.g., PersistentTokenStateService).
Then, the referencing code can check (tokenStateService instanceof PersistentTokenStore).
Perhaps, the persistent implementations could extend an abstract PersistentTokenStateService class, which itself extends DefaultTokenStateService.

Reply:
This works too and is a good idea. I submitted a new PS.